onnxruntime/contrib_ops/cuda/activation/activations_impl.cu

```diff
@@ -43,6 +43,13 @@ struct OP_Gelu : public CtxGelu {
   }
 };
 
+template <>
+struct OP_Gelu<half> : public CtxGelu {
+  __device__ __inline__ half operator()(const half& a) const {
+    return static_cast<half>(_Gelu(static_cast<float>(a)));
```

Comment:
Change the Gelu half kernel will impact performance (it adds two data type conversion, even though latency impact might be small). 

If user prefer better accuracy, they will use fp32 version of Gelu anyway. If they need performance, they would prefer the original implementation.

Reply:
As mentioned in the PR description, the perf impact is very small from Nsight Compute. The kernel for Gelu and GeluGrad will call _Normcdf and _Exp, and actually the half compute for these 2 functions a also cast data to float for compute and then cast back, so the cast is there anyway. Below are the comparison from Nsight Compute (baseline is the old impl):

Gelu:
![image](https://user-images.githubusercontent.com/11661208/174515779-ff19f43c-9c80-4344-8009-42a4f7b140a7.png)

GeluGrad:
![image](https://user-images.githubusercontent.com/11661208/174515852-c43e23f5-592c-4061-a5c1-2aa8e88a9f4c.png)